Fix #6154: Make SolanaWeb3 interface not accessible by Pages

Brandon-T opened the pull request:
## Security Review
- https://github.com/brave/security/issues/1040

## Summary of Changes
- Hide SolanaWeb3 in anonymous namespace variable only known to the Swift code by name.

The way the code works is that it generates a random UUIDv4 (we call this the namespace).
Then we inject JS code into that namespace like:

```javascript
// Global variable with random namespace name.
// Pages cannot enumerate variables so it isn't possible to know that this variable exists
// unless the page bruteforces all UUIDv4's and guesses this variable name.
// A 1 in 2.71 quintillion probability.

let W02C00302CE624DAEAD70FDEE19594856;

__firefox__.execute(function($, $Object) {
    // Pages cannot access local scope variables or code!
    inject_our_SolanaWeb3_Javascript_Here_as_LOCAL_scope();

    // Initialize our namespace to a hardened dictionary
    W02C00302CE624DAEAD70FDEE19594856 = $({
        "solanaWeb3": $(solanaWeb3)  // Hardened solanaWeb3 local scope object
    });
});
```

Now whenever our code needs to make calls to `solanaWeb3`, we have to do:
```javascript
W02C00302CE624DAEAD70FDEE19594856.solanaWeb3.WHATEVER_HERE

// Example:
let pubKey = new W02C00302CE624DAEAD70FDEE19594856.solanaWeb3.PublicKey(...);

return pubKey;  // Maybe $(pubKey) to harden it? But not sure if objects should be mutable.
```

Any page attempting to do `new solanaWeb3.PublicKey` will find that it doesn't exist and can't be accessed since it is not in global scope. The page would have to know our UUID generated variable name.
We could proxy all calls to our `W02C00302CE624DAEAD70FDEE19594856.solanaWeb3` as well, so that all returned objects/functions/values are immutable. Just not sure that this is needed or WHY.

<!-- Enter a ticket number for this PR, create a new one if it is not there yet. -->
This pull request fixes #6154

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

Brandon-T committed:
Hide SolanaWeb3 in anonymous namespace variable only known to the Swift code by name.

Brandon-T committed:
Inject wallet script at document start for now.

StephenHeaps reviewed (commented):
Functionality wise working great, thanks 👍. Unable to see `solanaWeb3` in Safari debugger . Couple small comments to address before merging but otherwise lgtm.

Brandon-T committed:
Convert Object Literals and Array Literals to Object.create and Array.of.

Brandon-T committed:
Secure SolanaWeb3 Script

Brandon-T committed:
Freeze objects returned to the page

Brandon-T committed:
Freeze all the solana objects returned to the page.
Do not recursively freeze transaction yet as it does some stuff internally and assigns to `this` internally.

Remove wallet handler as it isn't used and was experimental.

StephenHeaps reviewed (commented):
Everything working as expected for me. Just need to remove the unused `WalletHandler.swift` file.

StephenHeaps reviewed (changes requested):
Resolved in [0ff753b](https://github.com/brave/brave-ios/pull/6264/commits/0ff753b5a1880b1c8cfe42894bf14d808e12966c).

~Spoke too soon, _success_ flow for `Sign Transaction` & `Sign All Transactions (single)` are failing for me now; seems related to `solanaWeb3.Transaction` which both of `signTransaction` & `signAllTransactions` requests should return.~

1. Open pwgoom.csb.app
2. Connect and tap `Sign Transaction`
3. Tap `Continue`
4. Tap `Sign`

https://user-images.githubusercontent.com/5314553/203433063-5b423176-f6b2-4bf2-8fbf-6f3f7b0cd93f.mp4

Brandon-T committed:
Freeze Transactions.
Do not freeze `BN` (Big Number)! This is because `BN.prototype._strip` mutates `this` and will cause errors with transactions!

nuo-xu reviewed (approved):
PR lgtm. Tested it with test dapp. all the requests look and being handled correctly.

Brandon-T committed:
Fix toString to work with all object types and fixed regression where toString would not override properly

Brandon-T committed:
Access the solanaWeb3 namespace on initialization.
Fix order of injection so we can do proper exception handling if injection failed.

stoletheminerals reviewed (approved):
Just one minor logging issue. `SolanaWeb3` looks secure now. Though some `solana` properties are overwritable, I believe we don't want to freeze them for the same reason we don't freeze `ethereum` object.
Looks good, amazing job @Brandon-T

Brandon-T committed:
Remove unnecessary function
